hudi-flink/src/test/java/org/apache/hudi/utils/TestStringToRowDataConverter.java

```diff
@@ -98,10 +108,30 @@ void testRowDataToAvroStringToRowData() {
     final String[] recordKeys = KeyGenUtils.extractRecordKeys(recordKey);
     Object[] convertedKeys = stringToRowDataConverter.convert(recordKeys);
 
-    GenericRowData converted = new GenericRowData(6);
-    for (int i = 0; i < 6; i++) {
+    GenericRowData converted = new GenericRowData(7);
+    for (int i = 0; i < 7; i++) {
       converted.setField(i, convertedKeys[i]);
     }
     assertThat(converted, is(rowData));
   }
+
+  private TimestampData stringToTimestampData(String timestamp) {
+    return TimestampData.fromLocalDateTime(LocalDateTime.parse(timestamp));
+  }
+
+  private long timestampDataToLong(TimestampData timestamp, ChronoUnit unit) {
+    final LocalDateTime now = timestamp.toLocalDateTime();
+    final ZoneOffset offset = ZoneOffset.systemDefault().getRules().getOffset(LocalDateTime.now());
+    switch (unit) {
+      case MILLIS:
+        return TimeUnit.SECONDS.toMillis(now.toEpochSecond(offset))
+            + now.toInstant(offset).getLong(ChronoField.MILLI_OF_SECOND);
```

Comment:
Can we convert the `TimestampData` to `Instant` first then to the long then ? It is more concise i think.